app/models/playlist.rb

```diff
@@ -128,15 +129,21 @@ def self.latest(limit = 5)
   end
 
   def ensure_private_if_less_than_two_tracks
-    self.private = true if !is_favorite? && (tracks_count < 2)
+    self.private = true if !is_favorite? && !can_be_public?
```

Comment:
Do you know why this wasn't working before? It doesn't look like the logic actually changed (though I like it being broken out!)

Reply:
It was working, just needed this same check to ensure we are not allowing publishing if there is less than two tracks.
It might be overkill (considering that you said that this feature is disabled at a view level and we are not allowed to publish a playlist if there is less than 2 tracks), but it doesn't hurt to have a service level check.

Reply:
Ah cool, didn't realize it was in use in more than one place.